tools/alerting/src/main.rs

```diff
@@ -0,0 +1,514 @@
+        check_balance(context, balance)?;
+    }
+
+    Ok(())
+}
+
+fn check_balance(
+    context: &AlertingContext,
+    balance_config: &BalanceAlertingConfig,
+) -> anyhow::Result<()> {
+    let address = Pubkey::from_str(&balance_config.address)?;
+    let balance = context.rpc_client.get_balance(&address)?;
+
+    let min_balance_lamports = (balance_config.min_balance * 10u64.pow(9) as f64).floor() as u64;
+
+    info!(
+        "Balance for account {} is {}, min balance is {}",
+        address,
+        balance as f64 / 10u64.pow(9) as f64,
+        balance_config.min_balance
+    );
+
+    if balance < min_balance_lamports {
+        send_balance_alert(context, balance_config, balance)?;
+    } else {
+        clear_balance_alert(context, &address)?;
+    }
+
+    Ok(())
+}
+
+fn check_marginfi_group(
+    context: &AlertingContext,
+    group: &MarginfiGroupAlertingConfig,
+) -> anyhow::Result<()> {
+    let marginfi_program_id = Pubkey::from_str(&context.config.marginfi_program_id)?;
+    let group_address = Pubkey::from_str(&group.address)?;
+    let bank_accounts = context.rpc_client.get_program_accounts_with_config(
+        &marginfi_program_id,
+        RpcProgramAccountsConfig {
+            filters: Some(vec![RpcFilterType::Memcmp(Memcmp::new_raw_bytes(
+                8 + size_of::<Pubkey>() + size_of::<u8>(),
+                group_address.to_bytes().to_vec(),
+            ))]),
+            account_config: RpcAccountInfoConfig {
+                encoding: Some(UiAccountEncoding::Base64),
+                ..Default::default()
+            },
+            with_context: None,
+        },
+    )?;
+
+    let banks = bank_accounts
+        .into_iter()
+        .map(|(address, account)| {
+            let data = account.data.as_slice();
+            bytemuck::try_from_bytes::<Bank>(&data[8..])
+                .cloned()
+                .map(|b| (address, b))
+        })
+        .collect::<Result<Vec<(Pubkey, Bank)>, _>>()
+        .map_err(|e| anyhow::anyhow!(e))?;
+
+    info!("Found {} banks in group", banks.len());
+
+    let switchboard_v2_oracles = banks
+        .iter()
+        .filter(|(_, bank)| bank.config.oracle_setup == OracleSetup::SwitchboardV2)
+        .collect::<Vec<_>>();
+    // Pyth legacy is deprecated
+    let _pyth_oracles = banks
+        .iter()
+        .filter(|(_, bank)| bank.config.oracle_setup == OracleSetup::PythLegacy)
+        .collect::<Vec<_>>();
+    let pyth_push_oracles = banks
+        .iter()
+        .filter(|(_, bank)| bank.config.oracle_setup == OracleSetup::PythPushOracle)
+        .collect::<Vec<_>>();
+    let switchboard_pull_oracles = banks
+        .iter()
+        .filter(|(_, bank)| bank.config.oracle_setup == OracleSetup::SwitchboardPull)
+        .collect::<Vec<_>>();
+
+    check_switchboard_v2_oracles(context, &switchboard_v2_oracles)?;
+    check_pyth_push_oracles(context, &pyth_push_oracles)?;
+    check_switchboard_pull_oracles(context, &switchboard_pull_oracles)?;
```

Comment:
Nit: we don't really care if pull oracles are stale.

Reply:
Right now we are cranking all of them, so this is an alert if those fail